linter/lints/common.go

```diff
@@ -35,3 +37,39 @@ func GetExtWithOID(exts []pkix.Extension, oid asn1.ObjectIdentifier) *pkix.Exten
 	}
 	return nil
 }
+
+// ReadOptionalASN1BooleanWithTag attempts to read and advance incoming to
+// search for an optional DER-encoded ASN.1 element tagged with the given tag.
+// Unless outExpectedTagPresent or outPresent are nil, it stores whether the tag
+// was found in outExpectedTagPresent and an element with the tag was found in
+// outPresent, otherwise each boolean will take the default value. It reports
+// whether all reads were successful.
+func ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, out *bool, tag cryptobyte_asn1.Tag, defaultValue bool) bool {
+	if out != nil {
+		*out = defaultValue
+	}
```

Comment:
We should only set `*out` if we've succeeded and are going to return true. So this should move down below the call to `ReadOptionalASN1`.